include/maybe.hrl

```diff
@@ -70,6 +70,25 @@
              ___Err
          end)).
 
+-define(liftM(F, A1),
+        s2_maybe:liftM(F, [A1])).
+-define(liftM(F, A1, A2),
+        s2_maybe:liftM(F, [A1, A2])).
+-define(liftM(F, A1, A2, A3),
+        s2_maybe:liftM(F, [A1, A2, A3])).
+-define(liftM(F, A1, A2, A3, A4),
+        s2_maybe:liftM(F, [A1, A2, A3, A4])).
+-define(liftM(F, A1, A2, A3, A4, A5),
+        s2_maybe:liftM(F, [A1, A2, A3, A4, A5])).
+-define(liftM(F, A1, A2, A3, A4, A5, A6),
+        s2_maybe:liftM(F, [A1, A2, A3, A4, A5, A6])).
+-define(liftM(F, A1, A2, A3, A4, A5, A6, A7),
+        s2_maybe:liftM(F, [A1, A2, A3, A4, A5, A6, A7])).
+-define(liftM(F, A1, A2, A3, A4, A5, A6, A7, A8),
+        s2_maybe:liftM(F, [A1, A2, A3, A4, A5, A6, A7, A8])).
+-define(liftM(F, A1, A2, A3, A4, A5, A6, A7, A8, A9),
+        s2_maybe:liftM(F, [A1, A2, A3, A4, A5, A6, A7, A8, A9])).
```

Comment:
Why is this good? Instead of just defining like this perhaps.

```suggestion
-define(liftM(F, Maybes),
        s2_maybe:liftM(F, Maybes)).
```

Reply:
you're way too quick Jimmy 😅 there's definitely room for improvement, let me think about that

Reply:
I made some changes. the `liftm` macro now wraps as arguments in thunks to support lazy evaluation. I don't think we can achieve that without explicitly writing clauses for different numbers of arguments here.